_posts/2021-03-12-automated-accessibility-testing-leveraging-github-actions-and-pa11y-ci-with-axe.md

```diff
@@ -0,0 +1,200 @@
+"pa11y-ci:sitemap": "pa11y-ci --sitemap http://localhost:4000/sitemap.xml --sitemap-exclude \"/*.pdf\""
+```
+
+* `start-detached`: Starts the webserver that will run Jekyll to be used for testing.
+* `pa11y-ci:sitemap`: Runs pa11y-ci tests using the sitemap and excludes PDFs.
+
+Add a JSON file named `.pa11yci` which allows you to configure pa11y-ci with [various options](https://github.com/pa11y/pa11y#configuration). Here is our file:
+
+```json
+{
+  "defaults": {
+    "concurrency": 4,
+    "standard": "WCAG2AA",
+    "runners": ["axe"]
+  }
+}
+```
+
+* `concurrency`: The number of tests that should be run in parallel.
+* `standard`: The accessibility standard to use: Section508, WCAG2A, WCAG2AA (default), WCAG2AAA.
+* `runners`: The test runners to use: axe (run tests using aXe-core), htmlcs (default,  run tests using HTML CodeSniffer).
+
+That's it. You can run this setup locally using npm, and you will see the following output (truncated for brevity):
+
+```bash
+dmundra in ~/workspace/accessibility/accessibility on branch main > npm run start-detached
+
+> start-detached
+> bundle exec jekyll serve --detach
+
+Configuration file: /Users/dmundra/workspace/accessibility/accessibility/_config.yml
+            Source: /Users/dmundra/workspace/accessibility/accessibility
+       Destination: /Users/dmundra/workspace/accessibility/accessibility/_site
+ Incremental build: disabled. Enable with --incremental
+      Generating... 
+                    done in 8.217 seconds.
+ Auto-regeneration: disabled when running server detached.
+    Server address: http://127.0.0.1:4000
+Server detached with pid '14850'. Run `pkill -f jekyll' or `kill -9 14850' to stop the server.
+dmundra in ~/workspace/accessibility/accessibility on branch main > npm run pa11y-ci:sitemap
+
+> pa11y-ci:sitemap
+> pa11y-ci --sitemap http://localhost:4000/sitemap.xml --sitemap-exclude "/*.pdf"
+
+Running Pa11y on 49 URLs:
+ > http://localhost:4000/guide/glossary - 0 errors
+ > http://localhost:4000/guide/introduction - 0 errors
+ > http://localhost:4000/guide/history - 0 errors
+ > http://localhost:4000/guide/design - 0 errors
+...
+
+✔ 49/49 URLs passed
+```
+
+As you can see, at the time of this writing we have all our pages passing! To see examples of us fixing reported accessibility issues check out [https://github.com/CivicActions/accessibility/pull/128](https://github.com/CivicActions/accessibility/pull/128).
+
+Note: we have also noticed that pa11y-ci intermittently throws an error when trying to test many pages. Here is an open issue to find a solution to that: [https://github.com/pa11y/pa11y-ci/issues/128](https://github.com/pa11y/pa11y-ci/issues/128).
+
+## Running pa11y-ci in a GitHub action
+Now that we have it running locally, here is how we configure it to run every time we submit a pull request in GitHub.
+
+Add a YML file to `.github/workflows` folder. We called our version `pa11y.yml`. Here is our file:
+
+{% raw %}
```

Comment:
Liquid was converting the GitHub variables below. Fix from https://github.com/Shopify/liquid/wiki/Liquid-for-Designers#raw